homeassistant/components/ezviz/__init__.py

```diff
@@ -1 +1,147 @@
-"""Support for Ezviz devices via Ezviz Cloud API."""
+"""Support for Ezviz camera."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyezviz.client import EzvizClient
+from requests import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ACC_PASSWORD,
+    ACC_USERNAME,
+    ATTR_CAMERAS,
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+from .coordinator import EzvizDataUpdateCoordinator
+
+CAMERA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
+)
+
+EZVIZ_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Optional(ACC_USERNAME): cv.string,
+            vol.Optional(ACC_PASSWORD): cv.string,
+            vol.Optional(CONF_REGION, default=DEFAULT_REGION): cv.string,
+            vol.Optional(ATTR_CAMERAS, default={}): {cv.string: CAMERA_SCHEMA},
+        }
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: EZVIZ_SCHEMA}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+PLATFORMS = "camera"
+
+
+async def async_setup(hass: HomeAssistantType, config: dict) -> bool:
+    """Set up the Ezviz integration."""
+    hass.data.setdefault(DOMAIN, {})
+    conf = config.get(DOMAIN, EZVIZ_SCHEMA({}))
```

Comment:
H
There's no existing config section under the ezviz domain key. We should import the configuration YAML from the camera platform into a config flow instead.